typewriter/fixes/fixer_utils.py

```diff
@@ -105,6 +211,118 @@ def is_import_stmt(node):
     root.insert_child(insert_pos, Node(syms.simple_stmt, children))
 
 
+# -------------- TYPE_CHECKING LOGIC ------------------ #
+
+
+def _new_type_check_with_import(package, name, root, insert_pos):
+    # type: (Optional[str], str, Node, int) -> None
+    """
+    Inserts a new TYPE_CHECKING block containing a new import statement for package and name
+
+    Parameters
+    -----------
+    package : Optional[str]
+    name : str
+    root : Node
+    insert_pos : int
+    """
+    # [Grammar]
+    # if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)* ['else' ':' suite]
+    type_check_node = Node(syms.if_stmt,
+                           [Leaf(1, 'if'),
+                            Node(syms.power, [Leaf(1, 'typing'),
+                                              Node(syms.trailer, [Leaf(23, '.'),
+                                                                  Leaf(1, 'TYPE_CHECKING')])],
+                                 prefix=" "),
+                            Leaf(11, ':'),
+                            # [Grammar]
+                            # suite: simple_stmt | NEWLINE INDENT stmt+ DEDENT
+                            Node(syms.suite, [Leaf(token.NEWLINE, '\n'),
+                                              Leaf(token.INDENT, '    '),
+                                              Leaf(token.DEDENT, '')])])
+
+    children = [_generate_import_node(package, name), Newline()]
+    # We can just hardcode the correct insert position since we just created the typing block
+    type_check_node.children[3].insert_child(2, Node(syms.simple_stmt, children))
+    root.insert_child(insert_pos, type_check_node)
+    # Make sure to import typing.TYPE_CHECKING just before using
+    import_type_checking = [_generate_import_node(None, 'typing.TYPE_CHECKING'), Newline()]
+    root.insert_child(insert_pos, Node(syms.simple_stmt, import_type_checking))
+
+
+def create_type_checking_import(package, name, node):
+    # type: (str, str, Node) -> None
+    """
+    Create import statement of the form `from <package> import <name>` within a TYPING_CHECK
+    block
+
+    Parameters
+    -------------
+    package : str
+    name : str
+        Name of type being imported
+    node : Node
+    """
+
+    def is_type_checking_decl(node):
+        # [Grammar]
+        # if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)* ['else' ':' suite]
+        if not node.type == syms.if_stmt:
+            return False
+        stmt = str(node.children[1]).strip()
+        if stmt in ('typing.TYPE_CHECKING', 'TYPE_CHECKING'):
+            return True
+
+        return False
+
+    root = find_root(node)
+
+    # figure out where to insert the new import.  First try to find
+    # the first import and then skip to the last one.
+    insert_pos = 0
```

Comment:
Instead of `insert_pos = 0`, we should be doing something like `type_checking_suite = None`.

We then search for the suite (like you do below, with `is_type_checking_decl`), or else create it (with `_new_if_type_checking_suite` - which is the pared down version of `_new_type_check_with_import` - see note above).

Once we've either found or created `type_checking_suite`, the code from that point forward is the same.